crates/router/src/connector/stripe.rs

```diff
@@ -932,13 +933,15 @@ impl api::IncomingWebhook for Stripe {
     fn get_webhook_object_reference_id(
         &self,
         request: &api::IncomingWebhookRequestDetails<'_>,
-    ) -> CustomResult<String, errors::ConnectorError> {
+    ) -> CustomResult<ObjectReferenceId, errors::ConnectorError> {
         let details: stripe::StripeWebhookObjectId = request
             .body
             .parse_struct("StripeWebhookObjectId")
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
-        Ok(details.data.object.id)
+        Ok(ObjectReferenceId::PaymentId(
+            api_models::payments::PaymentIdType::ConnectorTransactionId(details.data.object.id),
+        ))
```

Comment:
Need logic to detect type of webhook and return appropriate ObjectReferenceId as opposed to always returning PaymentId

Reply:
Refund Flow is not yet implemented for this connector